Formula/coredns.rb

```diff
@@ -0,0 +1,83 @@
+class Coredns < Formula
+  desc "DNS server that chains plugins"
+  homepage "https://coredns.io/"
+  url "https://github.com/coredns/coredns/archive/v1.6.9.tar.gz"
+  sha256 "e100a946d5d60d936cb313bdb04fd96162d5a7229e08887ad497660acee4b36f"
+  head "https://github.com/coredns/coredns.git"
+
+  depends_on "go" => :build
+
+  def default_config
+    <<~EOS
+      . {
+        hosts {
+          fallthrough
+        }
+        forward . 1.1.1.1 1.0.0.1 8.8.8.8 8.8.4.4 9.9.9.9 149.112.112.112
+        cache
+        errors
+        log
+      }
+    EOS
+  end
```

Comment:
If it doesn't ship with a default config we shouldn't add one.

Reply:
Thank you for the advice @SMillerDev . I would like to know how familiar are you with CoreDNS before making the changes you are suggesting. The program in itself will work without a configuration file —[as you can see in the official Dockerfile](https://github.com/coredns/coredns/blob/ed1f42cf563e60068cadb96155423942917d0d47/Dockerfile)— but any DNS query will result in a loopback because it doesn’t have any knowledge of external DNS zones. Please refer to the example below:

```
# Terminal ONE
$ coredns
.:53
CoreDNS-1.6.9
darwin/amd64, go1.14.1, 1766568
[INFO] 127.0.0.1:55180 - 5564 "A IN google.com. udp 39 false 4096" NOERROR qr,aa,rd 88 0.000185024s
```

```
# Terminal TWO
$ dig +nocomments google.com @127.0.0.1 -p 1053
; <<>> DiG 9.10.6 <<>> +nocomments google.com @127.0.0.1 -p 1053
;; global options: +cmd
;google.com.			IN	A
google.com.		0	IN	A	127.0.0.1
_udp.google.com.	0	IN	SRV	0 0 54963 .
;; Query time: 0 msec
;; SERVER: 127.0.0.1#1053(127.0.0.1)
;; WHEN: Sun Apr 26 12:11:18 PDT 2020
;; MSG SIZE  rcvd: 99
```

Here you can see I started the CoreDNS server without the `-conf`. By default, it runs on port `::53` and doesn’t forward the DNS queries to any external DNS server. In the other Terminal window I executed the `dig` command to inspect the DNS records for google.com but as you can see it returns `127.0.0.1` instead of a public IP address.

Reply:
That seems like a very good reason for upstream to make a default config file.

Reply:
I partially agree with you.

On one hand, installing CoreDNS without a good configuration file will create (not very obvious) problems to users who barely understand what a DNS server is and simply want to do some experimentation. On the other hand, I understand why the CoreDNS developers do not have a default configuration file, doing so will inherently make them look like endorsers of a specific DNS provider (Cloudflare, Google, IBM, etc) and people who hate these companies will immediately complain.

Merging this pull-request will also make Homebrew look like people are endorsing Cloudflare, Google and IBM (in that specific order according to the "forward" statement in the Corefile). I don’t want to start an unnecessary discussion around the privacy policies of these companies and why sending your DNS traffic to them is a bad idea.

I will leave the pull-request open for a few hours in case you or other Homebrew members want to add constructive feedback. I will eventually close the merge request knowing there is no way to install this program without anyone complaining. People can install use this Formula with their own Homebrew Tap and this pull-request will serve as a sort of documentation.

Reply:
This pull request looks fine except for the fact that homebrew generally doesn't patch anything if it can be avoided. I'd think it would be a shame if you close this pull request because of a config file. I don't really care what is in the file, homebrew shouldn't pretend to know better than upstream in what config to ship.

Reply:
I have removed the unofficial configuration from the Formula with [commit 34ab2b2](https://github.com/Homebrew/homebrew-core/pull/53755/commits/34ab2b27bc5381ff1898d29bf5e6ed3d18d0b9b0). Installations via Homebrew will require —just like installations from the official sources— a configuration file with the “forward” statement in order to make DNS queries return useful IP addresses. Meanwhile, every DNS query will return a loopback IPv4 address.